### PR TITLE
HW_38

### PR DESCRIPTION
**Pull request by tatianakashliak98**

### CODE REVIEW

File: my-app/src/components/Post/Post.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import "./style.css"
+
+
+
+const Post = ({ title, date, image, description, id }: IPost) => {
+
+   return (
+      <li className='post__container'>
+         <div className='post__date'>{date}</div>
+         {image ? <img className='post__img' src={image} alt="img" /> : ""}
```
Comment (AlexKolb97) [inline my-app/src/components/Post/Post.tsx]: We don't need to render an empty string. If image is null/undefined - then do it like this:

`{image && img}`

File: my-app/src/components/Post/PostBig/PostBig.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React, { FC } from 'react'
+import { IPost } from '../../PostList/PostList'
+import "./style.css"
+
+const PostBig: FC<IPost> = ({ title, date, image, description, id }) => {
+   return (
+      <li id={`${id}`} className='post__container-big'>
```
Comment (AlexKolb97) [inline my-app/src/components/Post/PostBig/PostBig.tsx]: Why using back quotes here?

`<li id={id} className='post__container-big'>`

File: my-app/src/components/Post/PostBig/PostBig.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+         {image ? <img className='post-big__img' src={image} alt="img" /> : ""}
```
Comment (AlexKolb97) [inline my-app/src/components/Post/PostBig/PostBig.tsx]: Same.

File: my-app/src/components/Post/PostMiddle/PostMiddle.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import React, { FC } from 'react'
+import { IPost } from '../../PostList/PostList'
+import "./PostMiddle.css"
+
+const PostMiddle: FC<IPost> = ({ title, date, image, id }) => {
+   return (
+
+      <div id={`${id}`} className='post__container-middle'>
```
Comment (AlexKolb97) [inline my-app/src/components/Post/PostMiddle/PostMiddle.tsx]: Same.

File: my-app/src/components/Post/PostMiddle/PostMiddle.tsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+         {image ? <img className='post__img-middle' src={image} alt="img" /> : ""}
```
Comment (AlexKolb97) [inline my-app/src/components/Post/PostMiddle/PostMiddle.tsx]: Same.

File: my-app/src/components/Post/PostSmoll/PostSmoll.tsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import React, { FC } from 'react'
+import { IPost } from '../../PostList/PostList'
+import "./style.css"
+
+const PostSmoll: FC<IPost> = ({ title, date, image, id }) => {
+   return (
+
+      <li id={`${id}`} className='post__container-smoll'>
+         {image ? <img className='post__img-smoll' src={image} alt="img" /> : ""}
```
Comment (AlexKolb97) [inline my-app/src/components/Post/PostSmoll/PostSmoll.tsx]: Same.

File: my-app/src/components/PostList/PostList.tsx [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+  author?: number,
+  description?: string,
+  id: number,
+}
+
+const PostList = () => {
+  const [posts, setPosts] = useState<IPost[]>([]);
+
+
+  const backFunc = () => {
```
Comment (AlexKolb97) [inline my-app/src/components/PostList/PostList.tsx]: Name of the function should be informative. `backFunc` doesn't seems to say much..

File: my-app/src/components/PostList/PostList.tsx [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+      .then((data) => {
+        setPosts(data.results);
+
+      })
+      .catch((err) => {
+        console.log(err.message)
+      });
+  };
+  useEffect(() => {
+    return backFunc
```
Comment (AlexKolb97) [inline my-app/src/components/PostList/PostList.tsx]: What is the point of returning `backFunc` here? Did you mean to call it?

**Review by AlexKolb97 — CHANGES_REQUESTED** [review]
Please fix the comments and resolve conflicts.

File: my-app/src/components/Post/PostBig/PostBig.tsx (hunk 2 above)
Comment (tatianakashliak98) [reply my-app/src/components/Post/PostBig/PostBig.tsx]: ругается, что id string , поэтому так

**Pull request merged**